### PR TITLE
rsa-pss-sigs-on-certificate-verify - be strict with receiving alerts

### PR DESCRIPTION
**Pull request by tomato42**
### Description

make sure that the other side does send alerts and doesn't just close
connection on errors

### Motivation and Context

increase strictness of checks – no point in checking alert descriptions if we ignore missing alert

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [x] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [x] OpenSSL
  - [x] NSS
  - [x] GnuTLS
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/490)

### CODE REVIEW

**Comment by ueno** [comment]
The problem is specific to the "rsa_pss_pss_sha256 signature in CertificateVerify with rsa_pkcs1_sha256 id" test: unlike other 2 tests, the server is not able to distinguish whether the signature is calculated using an inconsistent signature scheme or it is simply invalid.  In fact, it fails in very low level in signature verification:
```c
server:stderr:|<4>| HSK[0x16cfba0]: verify cert vrfy: using RSA-SHA256
server:stderr:|<3>| ASSERT: pk.c[_wrap_nettle_pk_verify]:1193
server:stderr:|<3>| ASSERT: pubkey.c[pubkey_verify_data]:2217
server:stderr:|<3>| ASSERT: pubkey.c[gnutls_pubkey_verify_data2]:1783
server:stderr:|<3>| ASSERT: tls-sig.c[_gnutls_handshake_verify_crt_vrfy12]:387
server:stderr:|<3>| ASSERT: cert.c[_gnutls_proc_cert_client_crt_vrfy]:1242
server:stderr:|<3>| ASSERT: handshake.c[handshake_server]:3514
```
I think this is the case of "a handshake cryptographic operation failed" where decrypt_error shall be used.

**Comment by tomato42** [comment]
hmm, you're right, but only if the certificate has rsaEncryption key; if it's rsassa-pss then we're back to where the check happens (TLS or PKIX level), I'll push an update in a moment

File: scripts/test-rsa-pss-sigs-on-certificate-verify.py [hunk 1]
```diff
@@ -621,6 +627,12 @@ def main():
             bad += 1
             failed.append(c_name)
 
+    print("Check handling of rsa-pss signatures")
+    print("Test should be run twice, onece with certificate with")
```
Comment (ueno) [inline scripts/test-rsa-pss-sigs-on-certificate-verify.py]: typo: s/onece/once/

**Comment by tomato42** [comment]
Thanks for the review!

**Pull request merged**